### PR TITLE
Add trigger highlights

### PR DESCRIPTION
**Pull request by peterszerzo**
Fixes https://linear.app/nlxai/issue/FRO-17/pulsing-highlights-on-journey-manager.

@jakub-nlx this removes `findActiveTriggers` from the public API as it becomes internal to the package. So we can improve on the implementation later without having the end user have to change anything. Lame 500ms setInterval-based implementation for now, but only triggers if highlights are enabled.

https://github.com/nlxai/sdk/assets/6738398/d40a1677-ba1c-47cb-933f-88b256eb13e5

### CODE REVIEW

File: packages/journey-manager/src/index.ts [hunk 1]
```diff
@@ -376,12 +326,73 @@ export const run = (props: RunProps): RunOutput => {
     return [];
   };
 
+  /**
+   * UI management
+   */
+
+  // TODO: type this more accurately
+  let uiElement: any;
```
Comment (jakub-nlx) [inline packages/journey-manager/src/index.ts:334]:
```suggestion
  let uiElement: JourneyManagerElement | undefined;
```

File: packages/journey-manager/src/index.ts [hunk 2]
```diff
@@ -376,12 +326,73 @@ export const run = (props: RunProps): RunOutput => {
...
+    };
+    let highlightInterval: NodeJS.Timeout | null = null;
+    if (props.ui.highlights === true) {
+      highlightInterval = setInterval(() => {
+        const highlightElements = findActiveTriggers("click").flatMap(
+          (activeTrigger) => activeTrigger.elements,
+        );
+        uiElement.highlightElements = highlightElements;
+      }, 500);
+    }
```
Comment (jakub-nlx) [inline packages/journey-manager/src/index.ts:380]: Why not mutation observer? In fact, we could probably even build a faster `matchesActiveTriggers` that would fit neatly into mutation observer callbacks.

File: packages/journey-manager/src/index.ts (hunk 1 above)
Comment (peterszerzo) [reply packages/journey-manager/src/index.ts:334]: I couldn't make it work... what the `any` helps with is that it foregoes having to type custom events.

File: packages/journey-manager/src/index.ts [hunk 3]
```diff
@@ -160,10 +160,6 @@ export interface RunOutput {
    * Stop running the journey, removing all event listeners
    */
   teardown: () => void;
-  /**
-   * Find active triggers on the page
-   */
-  findActiveTriggers: (eventType: ActiveTriggerEventType) => ActiveTrigger[];
```
[the remark this answers is not in the document]
Comment (peterszerzo) [reply packages/journey-manager/src/index.ts:166]: Isn't this an API that you didn't feel good about, preferring a subscription model instead? Just curious how you imagine the entire API surface of this package looking like.

From my end this was a pragmatic step.

File: packages/journey-manager/src/index.ts (hunk 1 above)
Comment (jakub-nlx) [reply packages/journey-manager/src/index.ts:334]: Ah ok.

**Review by jakub-nlx — APPROVED** [review]

File: packages/journey-manager/src/index.ts (hunk 2 above)
Comment (peterszerzo) [reply packages/journey-manager/src/index.ts:380]: Here it is: https://github.com/nlxai/sdk/issues/103

**Pull request merged**